### PR TITLE
Activity Log: link plugin and theme names to their page in Calypso

### PR DESCRIPTION
**Pull request by eliorivero**
~~Companion to D10275-code~~ It was committed. Provides support for plugins.
Companion to D10307-code. Provides support for themes.
Related #22369 

This PR introduces support for new plugin and theme object formatting. This results in plugin and theme names linked to their page in Calypso.

captura de pantalla 2018-02-16 a la s 20 44 22

In the case of themes, if the theme isn't in wordpress.com, it will be linked (and open in a new tab) to the URL defined in the `Theme URI` header found in the theme's stylesheet.

Also updated the style of the link since it was showing in italics and bold:
captura de pantalla 2018-02-19 a la s 15 45 42

and now it's consistent with plugin links:
captura de pantalla 2018-02-19 a la s 15 46 29

### CODE REVIEW

File: client/lib/notifications/note-block-parser.js [hunk 1]
```diff
@@ -117,6 +117,13 @@ const userNode = ( { id: userId, name, site_id: siteId } ) => ( {
 	userId,
 } );
 
+const pluginNode = ( { site_slug, slug, version } ) => ( {
+	type: 'plugin',
+	siteId: site_slug,
```
Comment (dmsnell) [inline client/lib/notifications/note-block-parser.js]: noted in D10275, but it's a good idea to clarify what `site_slug` here is. it's _not_ the `siteId` regardless so we shouldn't conflate those terms.

passing the `siteId` in the API response or passing both the `siteId` and the `slug`/`domain` is another reasonable way to resolve ambiguity. mainly we can ask, do we want to preserve the slug-representation of the site in the URL or is a site id good enough…

Comment (eliorivero) [reply client/lib/notifications/note-block-parser.js]: Right, the `client/components/notes-formatted-block/index.jsx` currently uses:
```
return <a href={ `/plugins/${ name }/${ siteId }` }>{ descent }</a>;
```
so I think I should update it and use `siteSlug` everywhere.

File: client/components/notes-formatted-block/index.jsx [hunk 2]
```diff
@@ -66,13 +79,17 @@ export const FormattedBlock = ( { content = {} } ) => {
...
+				if ( -1 !== themeUri.indexOf( 'wordpress.com' ) ) {
+					return <a href={ `/theme/${ themeSlug }/${ siteSlug }` }>{ descent }</a>;
+				}
+				return (
+					<a href={ themeUri } target="_blank" rel="noopener noreferrer">
+						{ descent }
+					</a>
+				);
+			}
+			return descent;
```
Comment (dmsnell) [inline client/components/notes-formatted-block/index.jsx]: early-abort would keep out the nesting and I think add to the legibility here

```js
if ( ! themeUri ) {
	return descent;
}

if ( ! /wordpress\.com/.test( themeUri ) ) {
	return <a hre…>;
}

return <a hr…>;
```

Comment (eliorivero) [reply client/components/notes-formatted-block/index.jsx]: Makes sense

**Pull request merged**